pandas/tests/series/methods/test_replace.py

```diff
@@ -512,3 +512,10 @@ def test_pandas_replace_na(self):
         result = ser.replace(regex_mapping, regex=True)
         exp = pd.Series(["CC", "CC", "CC-REPL", "DD", "CC", "", pd.NA], dtype="string")
         tm.assert_series_equal(result, exp)
+
+    def test_replace_regex_dtype(self):
+        # GH-48644
+        s = pd.Series(["0"])
+        exp = s.replace(to_replace="0", value=1, regex=False).dtype
```

Comment:
explictly create the expected and use `tm.assert_series_equal`

Reply:
Done

Reply:
No, you should create expected = Series(...) not using replace. And please don't use one letter variable names

Reply:
Done